pkg/server/singleprocess/service_pipeline_run.go

```diff
@@ -0,0 +1,44 @@
+package singleprocess
+
+import (
+	"context"
+
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	serverptypes "github.com/hashicorp/waypoint/pkg/server/ptypes"
+)
+
+func (s *Service) ListPipelineRuns(
+	ctx context.Context,
+	req *pb.ListPipelineRunsRequest,
+) (*pb.ListPipelineRunsResponse, error) {
+	if err := serverptypes.ValidateListPipelineRunsRequest(req); err != nil {
+		return nil, err
+	}
+
+	result, err := s.state(ctx).PipelineRunList(req.Pipeline)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.ListPipelineRunsResponse{
+		PipelineRuns: result,
+	}, nil
+}
+
+func (s *Service) GetPipelineRun(
+	ctx context.Context,
+	req *pb.GetPipelineRunRequest,
+) (*pb.GetPipelineRunResponse, error) {
+	if err := serverptypes.ValidateGetPipelineRunRequest(req); err != nil {
+		return nil, err
+	}
+
+	result, err := s.state(ctx).PipelineRunGet(req.Pipeline, req.Sequence)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.GetPipelineRunResponse{
+		PipelineRun: result,
+	}, nil
+}
```

Comment:
These new server functions should have tests! A new file like `serivce_pipeline_run_test.go`, shouldn't be too complicated at the moment since they're mostly getting data from the DB and sending it back.

Reply:
These tests are actually included in the `service_pipeline` test under `RunPipeline`, since running a pipeline kicks off a pipeline run, I tossed in the List & Get tests for pipeline runs in there.

Reply:
I think those were good to include there too!

It'd be good to have some basic tests for these server functions that are as simple as upserting some PipelineRun messages directly and doing Gets and Lists (like Get returns the expected result, upserting ~3 runs means List returns 3, that sort of thing). I'm also thinking the UI Bundle server functions will need tests and having this in place will make it easier to add tests for those functions later too.

Reply:
Addressed, but Github didn't pick it up.